libs/aws/langchain_aws/retrievers/bedrock.py

```diff
@@ -9,49 +9,82 @@
 from langchain_core.retrievers import BaseRetriever
 from typing_extensions import Annotated
 
+FilterValue = Union[Dict[str, Any], List[Any], int, float, str, bool, None]
+
+
+class Criterion(BaseModel):
+    """A model to encapsulate a filter criterion with a key and its value."""
+
+    key: str
+    value: FilterValue
```

Comment:
Criterion doesn't seem like the most intuitive or descriptive name for the filter. Can we stick with the Filter and FilterValue names. Rather than creating a model class, a type is sufficient here.

Reply:
You are right, it is a little overkill. I simplified it with a simple `Filter`.